onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selector_action_transformer.cc

```diff
@@ -46,19 +47,31 @@ void DropQDQNodesRules(SelectorActionRegistry& qdq_selector_action_registry) {
 
   std::unique_ptr<Action> drop_action_no_int16 = std::make_unique<MergeIntoTargetFixed>(
       std::vector<NodeAndMoveInfo>(moves));  // Copy before std::move(moves)
+  std::unique_ptr<Action> drop_action_no_int16_nor_nonpositive_scale = std::make_unique<MergeIntoTargetFixed>(
+      std::vector<NodeAndMoveInfo>(moves));  // Copy before std::move(moves)
   std::unique_ptr<Action> drop_action = std::make_unique<MergeIntoTargetFixed>(std::move(moves));
 
 #if !defined(ORT_MINIMAL_BUILD)
-  // Use a separate selector + action that disallows 16-bit types for MaxPool and Resize.
+  // Use a separate selectors & actions for MaxPool and Resize.
+  //
+  // They disallow 16-bit types for MaxPool and Resize:
   // int16 MaxPool is not supported by the ONNX specification.
   // int16 Resize is not supported by the ORT implementation (although allowed by ONNX).
+  //
+  // And cannot eliminate the QDQ for MaxPool if the scale is not positive, as a negative scale will change the ordering
+  // of the elements between quantized & de-quantized values.
   std::unique_ptr<NodeSelector> selector_disallow_16bit = std::make_unique<QDQ::DropQDQNodesSelector>(false);
   qdq_selector_action_registry.RegisterSelectorAndAction(drop_action_no_int16_name,
-                                                         {{"MaxPool", {12}},
-                                                          {"Resize", {}}},
+                                                         {{"Resize", {}}},
                                                          std::move(selector_disallow_16bit),
                                                          std::move(drop_action_no_int16));
 
+  std::unique_ptr<NodeSelector> selector_disallow_16bit_and_nonpositive_scale = std::make_unique<QDQ::DropQDQNodesSelector>(false, true, false);
+  qdq_selector_action_registry.RegisterSelectorAndAction(drop_action_no_int16_nor_nonpositive_scale_name,
+                                                         {{"MaxPool", {12}}},
+                                                         std::move(selector_disallow_16bit_and_nonpositive_scale),
+                                                         std::move(drop_action_no_int16_nor_nonpositive_scale));
```

Comment:
this new action also needs to be registered with the same name using RegisterAction() in the minimal build code path (other part of the ifdef for `defined(ORT_MINIMAL_BUILD)`).

Reply:
I believe this is done in
https://github.com/microsoft/onnxruntime/pull/21182/commits/023bbafa10e429af208a2c613d869ef9ee335085